### PR TITLE
Method max

### PR DESCRIPTION
**Pull request by LuisARodr**
This adds the `max(comparer)` method functionality to the framework.
Also includes tests for the method.

Description of the method:
> This method will find and return the maximum value on the collection, a null value will be returned for empty arrays. If not comparer is specified then it will try to evaluate the array elements as if they are numbers. The comparator should be a function that receives 2 parameters, valueA and valueB and shall return a negative number for valueA < valueB scenario, zero when valueA === valueB and finally a positive number when valueA > valueB.

### CODE REVIEW

File: Methods/ArrayExtension.js [hunk 1]
```diff
@@ -312,10 +312,63 @@ Array.prototype.sum = function(spec) {
     return sum;
 }
 
+/**
+ * Returns the maximum value on the collection or null if the array is empty,
+ * if comparer is not present it will evaluate the elements as if they where 
+ * numbers.
+ * @param {comparer} comparer - A function used to make the comparations, this
+ * param not being present means the comparations will be made as if every value
```
Comment (gustavostuff) [inline Methods/ArrayExtension.js]: comparison*

File: Methods/ArrayExtension.js [hunk 2]
```diff
@@ -312,10 +312,63 @@ Array.prototype.sum = function(spec) {
...
+
+ /**
+  * A function that returns negative when valueA < value B, zero when 
+  * valueA === valueB or positive when valueA > valueB.
+  * @callback comparer
+  * @param {*} valueA - The first value to compare.
+  * @param {*} valueB - The second value to compare.
+  * @returns {number} - Returns a number being negative when valueA is bigger,
+  * zero when both values are the same or positive when valueB is bigger.
+  */
```
Comment (gustavostuff) [inline Methods/ArrayExtension.js]: newline

**Review by gustavostuff — CHANGES_REQUESTED** [review]

File: Methods/ArrayExtension.js (hunk 1 above)
Comment (LuisARodr) [reply Methods/ArrayExtension.js]: Fixed on c9f6e13

File: Methods/ArrayExtension.js (hunk 2 above)
Comment (LuisARodr) [reply Methods/ArrayExtension.js]: Fixed on c9f6e13

**Review by gustavostuff — APPROVED** [review]

**Comment by LuisARodr** [comment]
Closing to merge into [dev](https://github.com/LuisARodr/array-extension/tree/dev)

**Pull request closed without merging**